depscan/cli.py

```diff
@@ -826,12 +829,39 @@ def main():
             except Exception:
                 pass
         if run_cacher:
-            LOG.info(
-                "About to download the vulnerability database from %s. This might take a while ...",
-                vdb_database_url,
-            )
             oras_client = oras.client.OrasClient()
-            paths_list = oras_client.pull(target=vdb_database_url, outdir=data_dir)
+            use_rafs_image = False
+            nydus_image_command = shutil.which("nydus-image", mode=os.X_OK)
+
+            if nydus_image_command is not None:
+                LOG.info(
+                    "About to download the vulnerability database from %s. This might take a while ...",
+                    vdb_rafs_database_url,
+                )
+                use_rafs_image = True
+                try:
+                    rafs_data_dir = tempfile.TemporaryDirectory()
+                    paths_list = oras_client.pull(target=vdb_rafs_database_url, outdir=rafs_data_dir)
+                    os.system(
+                        f"{nydus_image_command} unpack --blob {rafs_data_dir}/data.rafs --output {data_dir}/vdb.tar --bootstrap {rafs_data_dir}/meta.rafs"
+                    )
+                    with tarfile.open(f"{data_dir}/vdb.tar", 'r') as tar:
```

Comment:
Check if the tar file is present before extract?

Reply:
Ack.